src/components/CardList/CardList.tsx

```diff
@@ -0,0 +1,294 @@
+import React, { JSX, useState, useEffect } from 'react';
+import { useSearchParams, useParams } from 'react-router-dom';
+import Card from '../Card/Card';
+import Spinner from '../Spinner/Spinner';
+import styles from './CardList.module.css';
+
+interface PokemonSummary {
+  name: string;
+  url: string;
+}
+
+interface Pokemon extends PokemonSummary {
+  id: number;
+  types: { type: { name: string } }[];
+  sprites: { front_default: string };
+}
+
+interface PokemonDetails extends Pokemon {
+  abilities: { ability: { name: string } }[];
+  stats: { stat: { name: string }; base_stat: number }[];
+}
+
+interface CardListProps {
+  searchTerm: string;
+}
+
+function CardList({ searchTerm }: CardListProps): JSX.Element {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { page } = useParams<{ page: string }>();
+  const currentPage = parseInt(page || searchParams.get('page') || '1', 10);
+  const pokemonId = searchParams.get('pokemonId');
+  const itemsPerPage = 24;
+  const [pokemonList, setPokemonList] = useState<Pokemon[]>([]);
+  const [selectedPokemon, setSelectedPokemon] = useState<PokemonDetails | null>(
+    null
+  );
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isDetailsLoading, setIsDetailsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+  const [detailsError, setDetailsError] = useState<string | null>(null);
+  const [totalPages, setTotalPages] = useState<number>(1);
+
+  useEffect(() => {
+    async function fetchPokemon(term: string) {
```

Comment:
Consider moving data fetching logic into a separate function or custom hook to keep the component cleaner and make the code easier to test and reuse.